web3/_utils/contracts.py

```diff
@@ -312,6 +319,21 @@ def encode_transaction_data(
     return add_0x_prefix(encode_abi(w3, fn_abi, fn_arguments, fn_selector))
 
 
+def decode_transaction_data(
+    fn_abi: ABIFunction,
+    data: HexStr,
+    normalizers: Sequence[Callable[[TypeStr, Any], Tuple[TypeStr, Any]]] = None,
+) -> Dict[str, Any]:
+    # type ignored b/c expects data arg to be HexBytes
+    data = HexBytes(data)  # type: ignore
+    types = get_abi_input_types(fn_abi)
+    abi_codec = ABICodec(default_registry)
+    decoded = abi_codec.decode(types, HexBytes(data[4:]))
+    if normalizers:
+        decoded = map_abi_data(normalizers, types, decoded)
+    return named_tree(fn_abi["inputs"], decoded)
```

Comment:
I understand the reasoning for wanting all regular tuples to be named tuples but I think it's a bit confusing / misleading to convert it when the flag isn't set on the contract instance. My biggest worry is that as a user I don't know what's going to come out as a tuple type and what will be a `ABIDecodedNamedTuple`. I think it would be ideal if the flag was an on/off switch. Am I missing a key idea of this PR or does that seem like a reasonable assumption?

This would apply for `get_event_data()` as well as it looks like this PR makes it default behavior there as well.